lesson_6_roman_alex.py

```diff
@@ -0,0 +1,85 @@
+    ('FI', ['84 ', '87 ', '89 ', '90 ', '90 ', '92 ', '94 ', '94 ', '94 ']),
+    ('FR', ['76 ', '80 ', '82 ', '83 ', '83 ', '86 ', '86 ', '89 ', '90 ']),
+    ('HR', ['61 ', '66 ', '65 ', '68 ', '77 ', '77 ', '76 ', '82 ', '81 ']),
+    ('HU', ['63 ', '67 ', '70 ', '73 ', '76 ', '79 ', '82 ', '83 ', '86 ']),
+    ('IE', ['78 ', '81 ', '82 ', '82 ', '85 ', '87 ', '88 ', '89 ', '91 ']),
+    ('IS', ['93 ', '95 ', '96 ', '96 ', ': ', ': ', '98 ', '99 ', '98 ']),
+    ('IT', ['62 ', '63 ', '69 ', '73 ', '75 ', '79 ', '81 ', '84 ', '85 ']),
+    ('LT', ['60 ', '60 ', '65 ', '66 ', '68 ', '72 ', '75 ', '78 ', '82 ']),
+    ('LU', ['91 ', '93 ', '94 ', '96 ', '97 ', '97 ', '97 ', '93 ', '95 ']),
+    ('LV', ['64 ', '69 ', '72 ', '73 ', '76 ', '77 ', '79 ', '82 ', '85 ']),
+    ('ME', [': ', '55 ', ': ', ': ', ': ', ': ', '71 ', '72 ', '74 ']),
+    ('MK', [': ', '58 ', '65 ', '68 ', '69 ', '75 ', '74 ', '79 ', '82 ']),
+    ('MT', ['75 ', '77 ', '78 ', '80 ', '81 ', '81 ', '85 ', '84 ', '86 ']),
+    ('NL', ['94 ', '94 ', '95 ', '96 ', '96 ', '97 ', '98 ', '98 ', '98 ']),
+    ('NO', ['92 ', '93 ', '94 ', '93 ', '97 ', '97 ', '97 ', '96 ', '98 ']),
+    ('PL', ['67 ', '70 ', '72 ', '75 ', '76 ', '80 ', '82 ', '84 ', '87 ']),
+    ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 ', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+    ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+    ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+    ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
+]
+
+
+# A function to build the smaller dictionaries on coverages and coresponding years
+
+def coverage_by_year(year_list,coverage_list):
+    year = year_list[1]
+    intermediary_container = []
+    for i in range(len(year)):
+        matrix = ((year[i],coverage_list[i]))
+        d = dict([matrix])
+        intermediary_container.append(d)
+    return(intermediary_container)
+
+# A function to build the dictionary - makes use of the above function
+
+def arange_data_as_requested(description_by_year,initial_raw_dataset):
+    arranged_data={
+        country:coverage_by_year(description_by_year,coverage_list)
+        for country,coverage_list in initial_raw_dataset
+    }
+    return(arranged_data)
+
+
+# get_year_data(dataset, "2019")
+# >>> {'2019': [('Romania', 84), ('Germany', 95), ..., ('Latvia', 85)]}
+
+def get_year_data(data_set, requested_year):
+    intermediary_list_container = []
+    for country,coverage_list in data_set.items():
+        for i in coverage_list:
```

Comment:
- `i` can be `year_data`.

I am not sure I understand this correctly 

`coverage_list` is a list of dictionaries  with the following structure: {'year': 2011, 'coverage': 85}. If this is the case the for `for year,coverage in i.items():` since it has only one iteration 

You can do 
```
for country,coverage_list in data_set.items():
        for year_data in coverage_list:
                if year_data['year'] == requested_year:
```

Reply:
Agreed. But unfortunately my since my raw data looks as follow:  {RO [ {2011: "83"}, {2012: "94"}, ........] } this approach will not work for me.